sdks/java/core/src/main/java/org/apache/beam/sdk/util/IOChannelUtils.java

```diff
@@ -51,18 +62,117 @@
    *
    * <p>For example, when reading from "gs://bucket/path", the scheme "gs" is
    * used to lookup the appropriate factory.
+   *
+   * @throws RuntimeException if override is false and schemes are conflicting.
```

Comment:
@throws IllegalArgumentException if multiple {@link IOChannelFactory}s for the same scheme are detected.